concordium-std-derive/src/lib.rs

```diff
@@ -1907,8 +1907,10 @@ const RESERVED_ERROR_CODES: i32 = i32::MIN + 100;
 /// Creating custom enums for error types can provide meaningful error messages
 /// to the user of the smart contract.
 ///
-/// Note that at the moment, we can only derive fieldless enums and the return
-/// value is always [`None`][std::option::Option::None].
+/// For variants with fields, all fields are serialized into the buffer used in
+/// the return value, which is always [`Some`][std::option::Option::Some].
+/// For fieldless variants, the return value is always
+/// [`None`][std::option::Option::None].
```

Comment:
I think this comment should be expanded a bit to also include that this error will be returned from the contract.

There is a complication, the same as it is now, that the schema will not support parsing this error.

So we probably need to update the schema for this to be fully done.

Although we can probably just merge this as is, it is not going to be worse than what we have now.

Reply:
I've added some more information to the comment. If that is not what you meant, then please let me know 😊